phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -1588,15 +1593,27 @@ public GetVersionResponse call(MetaDataService instance)
                 }
             }
         }
-        if (isIncompatible) {
-            buf.setLength(buf.length()-1);
-            throw new SQLExceptionInfo.Builder(SQLExceptionCode.OUTDATED_JARS).setMessage(buf.toString()).build().buildException();
+        if (!(errorCode.toString().isEmpty())) {
+            if (Integer.parseInt(errorCode.toString()) == SQLExceptionCode.OUTDATED_JARS.getErrorCode()) {
+                errorMessage.setLength(errorMessage.length()-1);
+                throw new SQLExceptionInfo.Builder(SQLExceptionCode.OUTDATED_JARS).setMessage(errorMessage.toString()).build().buildException();
+            } else {
+                throw new SQLExceptionInfo.Builder(SQLExceptionCode.INCOMPATIBLE_CLIENT_SERVER_JAR).setMessage(errorMessage.toString()).build().buildException();
+            }
         }
+
         if (systemCatalogTimestamp < MIN_SYSTEM_TABLE_TIMESTAMP) {
             throw new UpgradeRequiredException(systemCatalogTimestamp);
         }
     }
 
+    private String displayClientandServerVersions(long serverJarVersion) {
```

Comment:
nit: Rename method to "getClientAndServerVersions" since we're not really printing to stdout or logging the versions for it to be "display<xyz>".